nestauk/asf_heat_pump_suitability
# 2 add LSOA col to EPC

Fixes #2 

# Description

Create basic pipeline and add functions required to generate enhanced EPC dataset with LSOA column joined from ONS postcode directory.
- add `base_getters.py`: base getters to get dataset from ZIP files
- add `get_datasets.py`: specific getters to get ONSPD dataset
- add `enhance_epc_functions.py`: functions to clean and prepare EPC postcode column and join EPC with ONSPD to add LSOA column
- add `run_script.py`: run script to generate enhanced EPC dataset with additional LSOA and MSOA columns
- add `requirements.txt`
- add params to config `base.yaml`
- add `schemas.py`: data schemas for importing datasets

# Instructions for Reviewer

In order to test the code in this PR you need to:

- clone repo and follow set up instructions in README
- `cd` to repo folder
- run: `python asf_heat_pump_suitability/pipeline/run_script.py --epc_path s3://asf-daps/lakehouse/processed/epc/deduplicated/processed_dedupl-0.parquet --save_output s3://asf-heat-pump-suitability/outputs/test.parquet`
- This should generate the `test.parquet` file in the above S3 location. The dataset in this file should be EPC (selected cols) + 5 cols from ONSPD: pcd, lsoa11, lsoa21, msoa11, msoa21.

You can test the same function in a notebook with this code:
```
from asf_heat_pump_suitability.pipeline import run_script
epc_path = "s3://asf-daps/lakehouse/processed/epc/deduplicated/processed_dedupl-0.parquet"
output_path = "s3://asf-heat-pump-suitability/outputs/test.parquet"
test = run_script.main(epc_path=epc_path, save_output=output_path)
```

In your review, please could you:
- check the code works for you
- general check for errors/bugs/mistakes in the code and in the logic
- check the output of the code is as expected and desired

# Checklist:
